app/services/submissions.server.ts

```diff
@@ -129,3 +129,10 @@ export const prepareSubmission = async (
   });
   return contractData;
 };
+
+/**
+ * Returns an array of SubmissionDoc for a given UserAddress
+ */
+export const searchUserSubmissions = async (address: string) => {
```

Comment:
I think we can consolidate with `searchSubmissions` if we add serviceProvider as one of the search params